src/server.js

```diff
@@ -173,8 +173,11 @@ function init(socket, client, token) {
 		socket.emit("init", {
 			active: client.activeChannel,
 			networks: client.networks,
-			token: token || ""
+			token: token || "",
+			settings: client.settings.options,
 		});
+
+		client.settings.bindToSocket(socket);
```

Comment:
Actually yes. It's not currently needed, but I prefer having the functionality than it being a limitation later on. As I mentioned in the PR text itself, I'm aiming for API-ready code so I try to avoid putting arbitrary limitations. There are a few limited cases in which having the server aware of some settings could be useful. It's not used strictly for settings persistance, but also just for the server to share settings in general. I'm thinking `/ignore` support, auto away and others.

There's a (one-time) check [here](https://github.com/thelounge/lounge/pull/275/files#diff-a7181c2b6058f7bc955aa13d1e2a1befR57) to not persist settings in public mode. I'm using `client.name` so if we eventually allow both public and private mode at the same time, it's ready.